apps/condo/domains/billing/schema/RegisterBillingReceiptsService.js

```diff
@@ -0,0 +1,456 @@
+
+    const receiptsToUpdate = []
+    const receiptsToAdd = []
+    const notChangedReceipts = []
+
+    receipts.forEach((item) => {
+        const receiptKey = getBillingReceiptKey(
+            {
+                ...item,
+                ...{ recipient: { tin: item.tin, iec: item.iec, bic: item.bic, bankAccount: item.bankAccount } } },
+        )
+
+        const receiptExists = Reflect.has(receiptsIndex, receiptKey)
+
+        if (!receiptExists) {
+            receiptsToAdd.push(item)
+        } else {
+            const existingReceiptByKey = receiptsIndex[receiptKey]
+
+            const existingToPay = new Big(existingReceiptByKey.toPay)
+            const newToPay = new Big(item.toPay)
+            const toPayIsEqual = existingToPay.eq(newToPay)
+
+            const existingServices = existingReceiptByKey.services
+            // if not specified, then it is null
+            const newServices = item.services ? item.services : null
+            const servicesIsEqual = _.isEqual(existingServices, newServices)
+
+            const existingToPayDetails = existingReceiptByKey.toPayDetails
+            // if not specified, then it is null
+            const newToPayDetails = item.toPayDetails ? item.toPayDetails : null
+            const toPayDetailsIsEqual = _.isEqual(existingToPayDetails, newToPayDetails)
+
+            const shouldUpdateReceipt = !toPayIsEqual || !servicesIsEqual || !toPayDetailsIsEqual
+
+            if (shouldUpdateReceipt) {
+                item.id = existingReceiptByKey.id
+                receiptsToUpdate.push(item)
+            } else {
+                notChangedReceipts.push(item)
+            }
+        }
+    })
+
+    const newReceipts = []
+    for (const item of receiptsToAdd) {
+        const billingReceiptGQLInput = convertBillingReceiptToGQLInput(item, propertiesIndex, accountsIndex)
+        const newReceipt = await BillingReceipt.create(context, billingReceiptGQLInput)
+        newReceipts.push(newReceipt)
+    }
+
+    const updatedReceipts = []
+    for (const item of receiptsToUpdate) {
+        const itemId = item.id
+        const billingReceiptGQLInput = convertBillingReceiptToGQLInput(item, propertiesIndex, accountsIndex)
+        const updatableItem = _.omit(billingReceiptGQLInput, ['context', 'id'])
+        const updatedReceipt = await BillingReceipt.update(context, itemId, updatableItem)
+        updatedReceipts.push(updatedReceipt)
+    }
+
+    return { createdReceipts: newReceipts, updatedReceipts: updatedReceipts, notChangedReceipts: notChangedReceipts }
+}
+
+const RegisterBillingReceiptsService = new GQLCustomSchema('RegisterBillingReceiptsService', {
+    types: [
+        {
+            access: true,
+            type: 'input RegisterBillingReceiptInput ' +
+                '{ ' +
+                    'importId: String! ' +
+
+                    'address: String! ' +
+                    'normalizedAddress: String ' +
```

Comment:
add TODO to migrate it to address service

Reply:
Done